src/crimson/os/seastore/segment_manager/block.cc

```diff
@@ -293,7 +293,15 @@ write_superblock(
     bufferptr(ceph::buffer::create_page_aligned(sb.block_size)),
     [=, &device](auto &bp)
   {
+    // Write the SEASTORE_SUPERBLOCK_SIGN at offset 0 before
+    // encoding the superblock. If the signature is added inside
+    // the block_sm_superblock_t structure (via DENC), it won't be
+    // written at offset 0 because DENC encoding add some metadata.
```

Comment:
If DENC encodes additional data, can we use encode/decode instead so that we could move this into encode/decode?
As @cyx1231st mentioned earlier encoding additional data without properly decoding it might break future users (i.e objectstore-tool).

Reply:
I had already discussed with @cyx1231st he agreed with patch.